.github/workflows/main.yml

```diff
@@ -0,0 +1,123 @@
+name: Data-broker
+
+# Triggered after a branch is created, a tag or a branch is pushed or a PR
+on:
+  create:
+  push:
+  pull_request:
+
+jobs:
+  npm-build:
+    name: npm Build
+    # If the commit message contains 'ci skip', it is not necessary to run this job
+    if: contains(github.event.head_commit.message, 'ci skip') == false
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        node-version: [10.16.3]
+    steps:
+
+    - name: checkout
+      uses: actions/checkout@v1
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: npm ci, build, test and check licenses
+      run: |
+        npm ci
+        npm run-script build
+        npm run-script build-tests
+        npm test
+        npm run check-licenses
+
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        file: coverage/lcov.info # Created when 'npm test' is executed
+        flags: Data-broker # Flags allow you to isolate coverage reports for different tests
+        name: data-broker # Custom name for the upload
+
+
+  docker-compose:
+    needs: npm-build
+    name: Docker Compose build
+    runs-on: ubuntu-18.04
+    steps:
+
+    - name: checkout
+      uses: actions/checkout@v2
+
+    - name: Build the stack and test
+      run: |
+        docker build -t ${GITHUB_REPOSITORY} . -f ./docker/Dockerfile
+        docker build -t dredd-tests . -f ./tests/Dockerfile
+        docker-compose -f ./tests/docker-compose.yml up -d kafka data-broker-redis
```

Comment:
Did you test when the dread (docker-compose) has some error?
Because I think It's important to see if some error happen in the dread the github actions return a status fail.